v1/src/main/java/com/google/cloud/teleport/templates/BulkCompressor.java

```diff
@@ -159,6 +159,18 @@ public interface Options extends PipelineOptions {
     ValueProvider<Compression> getCompression();
 
     void setCompression(ValueProvider<Compression> value);
+
+    @TemplateParameter.Text(
+        order = 5,
+        optional = true,
+        regexes = {"^[A-Za-z_0-9.]*"},
+        description = "Output filename suffix",
+        helpText =
+            "Output filename suffix of the files to write. Defaults to .bzip2, .deflate or .gz depending on the compression algorithm.")
+    @Required
```

Comment:
The Required here slightly contradicts `optional = true` from above. They are different constructs but may raise some confusion.

Could we please remove it for clarity before merging?

Thanks!

Reply:
This confused me too and I am actually not sure what `@Required` does here or why it seems required by the code, but without it the resulting template did not know the `outputFilenameSuffix` parameter:

> ERROR: (gcloud.dataflow.jobs.run) INVALID_ARGUMENT: (451b3c759e4a9769): The workflow could not be created. Causes: (451b3c759e4a994c): Found unexpected parameters: ['outputFilenameSuffix' (perhaps you meant 'outputFailureFile')]

Reply:
Ah, I see, my bad.

Can you test replacing it with `@Default.String("")` to see how it goes? 

(Reminder that you would have to replace the conditions on the `@ProcessElement` to also account for empty, perhaps `StringUtils.isNotEmpty(options.getOutputFilenameSuffix())`)

Thanks!

Reply:
Can you test replacing it with @Default.String("") to see how it goes? 

That wouldn't be a good solution since an empty string could be a desirable suffix value if you want to avoid the renaming at all.